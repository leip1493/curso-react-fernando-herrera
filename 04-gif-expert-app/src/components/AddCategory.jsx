import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({ target: { value } }) => {
    setInputValue(value);    
  };

  const handleSubmit = e => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type="text" onChange={handleInputChange} value={inputValue} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};
