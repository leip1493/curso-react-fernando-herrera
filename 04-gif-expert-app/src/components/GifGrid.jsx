import React from "react";
import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category} </h3>
      {loading && <p className="animate__animated animate__flash">Loading</p>}
      <div className="card-grid">
        {gifs.map(gif => (
          <GifGridItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
