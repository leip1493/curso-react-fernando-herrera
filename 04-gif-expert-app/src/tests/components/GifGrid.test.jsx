import React from "react";
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs"); // 1 - Falsea el llamado de la funci√≥n

describe("<GifGrid />", () => {
  const category = "Naruto";

  test("Debe renderizar exitosamente", () => {
    // Estado al primer renderizado del componente
    useFetchGifs.mockReturnValue({
      // 2 - Falsea el contenido de respuesta de el custom hook/funcion
      gifs: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar items cuando cargan los gifs useFetchGifs", () => {
    const gifs = [
      {
        id: 1,
        url: "https://localhost/imagen.jpg",
        title: "lorem ipsum"
      }
    ];
    // Estado luego del primer renderizado
    useFetchGifs.mockReturnValue({
      gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />);
    
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBeFalsy();
    // Valida que exista tantos GifGridItem como gifs consultados existan
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);

  });
});
