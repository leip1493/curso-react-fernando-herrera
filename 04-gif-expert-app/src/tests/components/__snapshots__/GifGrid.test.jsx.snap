// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> Debe renderizar exitosamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
